src/AvaloniaEdit/Text/TextLineRun.cs

```diff
@@ -59,19 +59,22 @@ public double Height
                     return box.Height;
                 }
                 
-                return GetDefaultLineHeight(FontSize);
+                return GetDefaultLineHeight(TextRun.Properties.FontMetrics);
             }
         }
 
-        public static double GetDefaultLineHeight(double fontSize)
+        public static double GetDefaultLineHeight(FontMetrics fontMetrics)
         {
-            // add 2pt for underline
-            return fontSize * HeightFactor + 2;
+            // add additional space for underline and line gap
+            return fontMetrics.LineHeight +
```

Comment:
LineHeight already includes the LineGap. We don't need to add the UnderlinePosition to the line height because the position is always within the LineHeight.

Reply:
@Gillibald if I don't add extra height the lines are very close. Let me show you some screenshots...

Reply:
Using just `fontMetrics.LineHeight`, makes lines to be too close compared with VSCode, for example.
![image](https://user-images.githubusercontent.com/501613/143769028-f9a975af-4344-4f24-9317-03fcd2099504.png)

What should I do, use an extra hardcoded margin?

Reply:
Adding those extra `1.740234375` px added by adding `fontMetrics.UnderlinePosition + fontMetrics.UnderlineThickness` makes AvaloniaEdit to look exactly like VSCode.

Reply:
Should I calculate that needed `1.740234375` extra height as a percentage of the line height?

Reply:
Something like this?
```
        public static double GetDefaultLineHeight(FontMetrics fontMetrics)
        {
            double factor = 1.15;
            return fontMetrics.LineHeight * factor;
        }
```

Reply:
This line is wrong https://github.com/AvaloniaUI/Avalonia/blob/master/src/Skia/Avalonia.Skia/FormattedTextImpl.cs#L557 so you need to add the LineGap on your own.

Reply:
@Gillibald, Any recommended calculation for the line gap?

Reply:
Something like this?
```
        public static double GetDefaultLineHeight(FontMetrics fontMetrics)
        {
            // FontMetrics doesn't include a lineGap to the total lineHeight
            // so we need to add the LineGap on our own
            double lineGap = fontMetrics.LineHeight * 0.15;
            return fontMetrics.LineHeight + lineGap;
        }
```

Reply:
Hmm then there must be something else wrong / different. A LineGap of zero is totally fine. As long as the underline renders just fine I don't see any issue. In general you can't compare the rendering results of two different text rendering stacks. There might be different rounding / scaling in place etc.

Reply:
Also it should be fine to add some extra space  to each line as you have done. This should rely on some fixed factor and the current fontSize.

Reply:
Adding an extra 15% of the line height works fine for different font sizes. I'll change that. Thank you!